sources/lib/util/options.mli

```diff
@@ -507,8 +527,12 @@ val output_native : unit -> bool
 val output_smtlib : unit -> bool
 (** Default to [false] *)
 
+(** [true] if the output format is set to [Why3] *)
+val output_why3 : unit -> bool
+(** Default to [false] *)
```

Comment:
These functions (all of the `output_lang`) should disappear, and instead the call-sites should be replaced with a match (or an equality test) on the result of  `input_format ()`.

Reply:
`output_format ()`*

Reply:
Well, it won't actually work if you want to do it outside of options since (=) is set to be of type int -> int -> int at the end of options.ml. Any equality test outside of options not involving integers won't work hence the need to define these flag getters here

Reply:
Well, in that case there are two good solutions (compared to the bad one show here):
- instead of using esuality, always use a match on the `output_format` which actually is reasonable and can help greatly future-proofing the code, using the exhaustive match check of the compiler.
- or, the option module can export a comparison and/or equality function on types where it might be useful

Reply:
I'm actually curious to know why this one is "bad" since we're just hiding behind functions the kind of outputs that are available and the way they're implemented. Your first solution would actually make the implementation visible everywhere and any change of it would mean changing it in every file while here it means only changing 3 functions. Your second option has the same problem. I kind of like the actual solution (which was implemented a long time ago)

Reply:
Well because basically, everywhere you use one of these, you should actually do something different, but still something for each language: basically, the output format (resp. the input format), should **only** be used to print (resp. parse/typecheck) differently based on the format choosen. Other considerations should have a dedicated logic (which can use the formats to decide some things), which has a name, documentation, and specified semantics, otherwise it's an unbeliveable mess to try and understand let alone modify code. This is the case currently, but it would be good if we could fix that when we touch the relevant code.

Reply:
Ah well, this has then nothing to do with the current PR and it would add noise for nothing then.

Reply:
Well, the current PR is a good time to fix that part and we should absolutely do it now, else we won't do it. Basically every time we touch a piece of code, if the modification is not time sensitive (such as fixing a soundness bug), we should clean up the related part of the code, else we'll be stuck with atrocious, unbearable code on which every modification is a nightmare because there is no doc or clear semantics.

For instance, imagine when we will add smtlib v3 which should release soon if I recall correctly. I expect the semantics to really not differ much between smtlib2 and smtlib3, at least on the fragment which is handled by alt-ergo. So basically, with things the way they are, everywhere there is a `output_smtlib2 ()`, one should add a `|| output_smtlib3 ()`, which is not only tiresome but also error prone, sinc one could very easily miss one such occurrence, and review on the github API will easily miss such missing occurrences. With pattern matches, it becomes a lot easier to spot everywhere this need to be updated.

Reply:
Done